nexus/db-queries/src/db/datastore/mod.rs

```diff
@@ -200,29 +200,53 @@ impl DataStore {
             .unwrap();
     }
 
-    // TODO-security This should be deprecated in favor of pool_authorized(),
-    // which gives us the chance to do a minimal security check before hitting
-    // the database.  Eventually, this function should only be used for doing
-    // authentication in the first place (since we can't do an authz check in
-    // that case).
-    fn pool(&self) -> &bb8::Pool<ConnectionManager<DbConnection>> {
-        self.pool.pool()
-    }
-
-    pub(super) async fn pool_authorized(
+    async fn pool_authorized(
```

Comment:
This is sorta the meat-and-potatoes of this PR:

- We no longer expose the raw pool to most of Nexus
- Instead, we encourage callers to access a connection, via one of the `pool_connection...` functions below
- Queries can be issued against individual connections

This is nice, because we **must** act on connections from a transaction context anyway, so doing this unifies the way we issue queries -- regardless of whether you're writing code within a transaction, you'll always be acting on a single connection now, and you'll only need to deal with connection pool errors when acquiring the connection.